packages/access-api/src/models/delegations.js

```diff
@@ -0,0 +1,137 @@
+import * as Ucanto from '@ucanto/interface'
+import {
+  delegationsToBytes,
+  bytesToDelegations,
+} from '@web3-storage/access/encoding'
+
+/**
+ * @typedef {import('@web3-storage/access/src/types').DelegationTable} DelegationRow
+ * @typedef {Omit<DelegationRow, 'inserted_at'|'updated_at'|'expires_at'>} DelegationRowUpdate
+ */
+
+/**
+ * @typedef Tables
+ * @property {DelegationRow} delegations
+ */
+
+/**
+ * @typedef {import("../types/database").Database<Tables>} DelegationsDatabase
+ */
+
+/**
+ * DelegationsStorage that persists using SQL.
+ * * should work with cloudflare D1
+ */
+export class DbDelegationsStorage {
+  /** @type {DelegationsDatabase} */
+  #db
+
+  /**
+   * @param {DelegationsDatabase} db
+   */
+  constructor(db) {
+    this.#db = db
+    // eslint-disable-next-line no-void
+    void (
+      /** @type {import('../types/delegations').DelegationsStorage} */ (this)
+    )
+  }
+
+  get length() {
```

Comment:
It's unusual (IMHO) for a property accessor to return a promise. Why not a function?

```suggestion
  async length() {
```

Reply:
originally this type started as a subset of `Array` and I did some TDD from there to implement the interface backed by sql, and that required affording for async. I figured it would make sense to adjust the names/types later in the context of a specific reviewer. Happy to change this to a function now. will do

Reply:
IMHO properties make sense for nouns like 'length' and functions make sense for methods. I'm gonna call the function count() (inspired by sql count(), mongo count(), etc). can change it again if you def want 'length()'

Reply:
renamed to `count(): Promise<bigint>` in https://github.com/web3-storage/w3protocol/pull/427/commits/022dc2167647dcbabb35b6ab0450dafd9c708a84